scipy/optimize/tests/test_cython_optimize.py

```diff
@@ -0,0 +1,93 @@
+"""
+test cython optimize zeros api functions
+"""
+
+from __future__ import division, print_function, absolute_import
+import numpy as np
+from scipy.optimize.cython_optimize import _zeros
+
+
```

Comment:
The testing being done here is very opaque.  It is not clear to me what function(s) is being used, what the args are, what the is bounding interval.  There are several levels of indirection involved, with no comments to guide  in case of a failure, or to validate what it is doing now.   Can you think about how to make this more readily transparent for reviewers and future maintainers?